guides-markdown/vps-setup.md

```diff
@@ -0,0 +1,338 @@
+```
+$ usermod -aG admin test
+$ usermod -aG sudo test
+```
+
+Now to login to the server you will need to tell SSH to use this username. For Linux/Mac this is automatic if your system username is the same as the username you create.
+
+```
+$ ssh test@fullnode
+```
+
+Once logged in, you can preface commands with `sudo` to run them as the root user or change into the root user
+
+```
+$ sudo apt-get install htop
+$ sudo su -
+```
+
+## Install Bcoin
+```
+$ npm install -g bcoin --production
+```
+
+### Command Line Interaction
+After bcoin installs we're going to store some configuration info and sensitive daat as environmental variables.
+
+```
+$ export BCOIN_API_KEY=hunter2
+$ export BCOIN_NETWORK=main
+$ export BCOIN_URI=http://localhost:8332
+$ bcoin --index-tx --index-address --daemon
+${PID}
+```
+
+Alternatively, you can assign http port and network using command line flags:
+
+```
+$ bcoin --index-tx --index-address --http-port=8332 --network=main
+```
+
+Here, PID is the process ID of bcoin. To stop bcoin you can issue `kill ${PID}` replacing `${PID}` with the number your terminal returns. If you forget the PID you can find it again with `ps -ef | grep bcoin`
+
+You can use the bcoin CLI tool out of the box. To view your node's status:
+
+```
+$ bcoin cli info --api-key=hunter2
+{
+  "version": "v1.0.0-beta.14",
+  "network": "main",
+  "chain": {
+    "height": 124767,
+    "tip": "0000000000003ae2272bb2ad89d11b3b3119536690de1bf3b3180d3e0df91d27",
+    "progress": 0.2604341435491337
+  },
+  "pool": {
+    "host": "2601:c6:c880:3020:1a5e:fff:febd:ff6",
+    "port": 8333,
+    "agent": "/bcoin:v1.0.0-beta.14/",
+    "services": "1001",
+    "outbound": 8,
+    "inbound": 0
+  },
+  "mempool": {
+    "tx": 0,
+    "size": 0
+  },
+  "time": {
+    "uptime": 450,
+    "system": 1517778800,
+    "adjusted": 1517778907,
+    "offset": 107
+  },
+  "memory": {
+    "total": 202,
+    "jsHeap": 35,
+    "jsHeapTotal": 70,
+    "nativeHeap": 131,
+    "external": 8
+  }
+}
+
+```
+
+For a complete list of CLI commands, check the documentation in [Github](https://github.com/bcoin-org/bcoin/blob/d67d089c53ec9a72bf967824e7dfc89d27d5354f/docs/CLI.md). The CLI supports node, wallet and has full RPC functionality.
+
+### Using Bcoin as a library
+
+You can use a tool like [forever](https://github.com/foreverjs/forever) to run your scripts that use bcoin as daemons. A benefit of forever is that it will automatically restart your script if it fails. It can detect infinite looping and ultimately stop the script if necessary.
+
+```
+$ git clone git@github.com:bcoin-org/bcoin.git
+$ cd bcoin
+$ forever start docs/Examples/fullnode.js
+warn:    --minUptime not set. Defaulting to: 1000ms
+warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
+info:    Forever processing file: docs/Examples/fullnode.js
+```
+
+Check your node with
+
+```
+$ bcoin cli info
+```
```

Comment:
Should probably use the new `bclient` package here: https://github.com/bcoin-org/bclient